server/e2e_test.go

```diff
@@ -0,0 +1,572 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/cashshuffle/cashshuffle/message"
+
+	"github.com/avast/retry-go"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	basicPoolSize = 3
+	someAmount    = uint64(100000000)
+	someVersion   = uint64(999)
+)
+
+var (
+	testTempKey int
+)
+
+// TestHappyShuffle simulates a complete shuffle
```

Comment:
NIT: period at the end?

Reply:
wew I was loose on these. I will clean them all up.

Reply:
26b8146

should be better